### ISSUE TITLE
Whitespace handling in `Content::plain_text`

### ISSUE THREAD

**Issue by mattfbacon** [typst/typst#2482]
I initially noticed that for code like this:

```typst
= My heading \
that goes on two lines!
```

it would be in the PDF outline as `My heading  that goes on two lines!`. Note the double space.

I traced this back to `Content::plain_text`. I think that this is not the correct output because if you were to join the two lines together by replacing `"\\\n"` with `""`, you would end up with only one space.

**Comment by jgpr-code** [comment]
I was just looking for another issue regarding content and whitespace and noticed that this one seems already to be fixed with typst 0.11.0 (2bf9f95d)

**Comment by jgpr-code** [comment]
![image](https://github.com/typst/typst/assets/14888393/8f5d5492-3d37-42e7-9e83-5163e749207b)
![image](https://github.com/typst/typst/assets/14888393/61d5187a-5dcf-4f06-9b95-edd5aeb8830d)

For me this does not look like that

**Comment by jgpr-code** [comment]
maybe you are using an older version of typst than me?

**Comment by mattfbacon** [comment]
Yeah the initial example was not entirely correct. Here was my test case:

```typst
#heading(level: 1, [My heading \
that goes on multiple lines])
```

**Comment by mattfbacon** [comment]
![Outline showing "My heading  that goes on ..." (note the double space between "heading" and "that")](https://github.com/typst/typst/assets/58113890/a300cb58-b934-4090-b908-27e3dbcebc1e)

This is the issue. Notice the double space.

**Comment by jgpr-code** [comment]
this does not explain how you get to that state.

It would be easier to have a minimal example that can reproduce your issue, as shown above from the information here it is still not possible to reproduce it.

I suspect that you have some other interaction going on which might or might not be an actual problem.

**Comment by mattfbacon** [comment]
The minimal example is as given above. Look at the outline in a PDF viewer capable of showing it. There is a double space in the title.

You can show this in code by using pymupdf.

```python
>>> import fitz
>>> doc = fitz.open('/tmp/a.pdf') # PDF generated from typst code given above.
>>> toc = doc.get_toc()
>>> print(toc)
[[1, 'My heading  that goes on multiple lines', 1]]
```

**Comment by jgpr-code** [comment]
Maybe there is a difference between a linebreak and an actual newline character in pdf?
At least I finally understand your issue, but I don't know enough about pdf or typst to be able to help you there.

In Typst you could try to use the newline character directly with unicode
```
#outline()

#heading(level: 1, [
  My heading \
  that goes on multiple lines
])

#heading(level: 1, [
  This heading\u{000A}goes on multiple lines
])
```

loading the resulting document with fitz code like this:

```
import fitz

pdf_document = "quick_tests.pdf"  # Replace with your actual PDF file path
doc = fitz.open(pdf_document)

page1 = doc.load_page(0)
page1text = page1.get_text()
print("Text from PDF: ", page1text)

toc = doc.get_toc()
print(toc)
```

gives me output like that:

```
(.venv) typst_issue>py test.py
Text from PDF:  Contents
My heading
that goes on multiple lines ..................................................................................................................................... 1
This heading
goes on multiple lines .............................................................................................................................................. 1
My heading
that goes on multiple lines
This heading
goes on multiple lines
```
